detectable/src/main/java/com/blackduck/integration/detectable/detectables/cargo/CargoDependencyTransformer.java

```diff
@@ -0,0 +1,92 @@
+package com.blackduck.integration.detectable.detectables.cargo;
+
+import com.blackduck.integration.bdio.graph.BasicDependencyGraph;
+import com.blackduck.integration.bdio.graph.DependencyGraph;
+import com.blackduck.integration.bdio.model.Forge;
+import com.blackduck.integration.bdio.model.dependency.Dependency;
+
+import com.blackduck.integration.bdio.model.dependency.ProjectDependency;
+import com.blackduck.integration.bdio.model.externalid.ExternalId;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+//TODO: Proposed CargoDependency Parser, Need to be implemented to parser
+
+public class CargoDependencyTransformer {
+
+    public DependencyGraph transform(JsonObject jsonObject) {
+        BasicDependencyGraph graph = new BasicDependencyGraph();
+        Map<String, Dependency> packageIdToDependency = new HashMap<>();
+        Set<String> allDependencies = new HashSet<>();
+
+        JsonArray packages = jsonObject.getAsJsonArray("packages");
+        for (JsonElement pkg : packages) {
+            JsonObject packageObj = pkg.getAsJsonObject();
+            String packageId = packageObj.get("id").getAsString();
+            String name = packageObj.get("name").getAsString();
+            String version = packageObj.get("version").getAsString();
```

Comment:
If you did go with the POJO approach it might make this code a bit easier to understand.